### DISCUSSION TITLE
Support optional ip parameter in announce query string

### DISCUSSION THREAD

**Discussion by coldwinds** [torrust/torrust-tracker#532] [Feature requests]
Hi,

Is the ip parameter could be supported as an option?

see here https://wiki.theory.org/BitTorrent_Tracker_Protocol

`
http://some.tracker.com:999/announce
?info_hash=12345678901234567890
&peer_id=ABCDEFGHIJKLMNOPQRST
&ip=255.255.255.255
&port=6881
&downloaded=1234
&left=98765
&event=stopped
`

Which will give a chance to reverse proxy to pass peer's ip by rewrite peer's query and add/overwrite an ip parameter to it.

Opentracker's Makefile also have a build option for this:

https://erdgeist.org/arts/software/opentracker/

`
By default opentracker will only allow the connecting endpoint's IP address to be announced. Bittorrent standard allows clients to provide an IP address in its query string. You can make opentracker use this IP address by enabling -DWANT_IP_FROM_QUERY_STRING.
`

It was previously posted in this issue https://github.com/torrust/torrust-tracker/issues/163#issuecomment-1836642956 , and @josecelano have some good ideas on it.

**Comment by tramix32** [comment]
This option would be very useful because it is currently not possible to seed from the same local network as the tracker and computer despite setting the external IP address in qbit.

**Reply by josecelano** [reply]
Hi, @tramix32. We are close (approximately 1 month) to releasing a new major version (3.0.0). After that, I would like to focus on prioritizing issues in the roadmap (after getting feedback from the community). I think it would be good to implement many of these small features that are easy to implement but very useful.

For example:

- [ ] New option `want_ip_from_query_string`. See <https://github.com/torrust/torrust-tracker/discussions/532#issuecomment-1836642956>.
- [x] Permanent keys. See <https://github.com/torrust/torrust-tracker/discussions/244>.
- [ ] Peer and torrents specific statistics. See <https://github.com/torrust/torrust-tracker/discussions/139>.

This fits into that category.

**Comment by SuspiciousDuck** [comment]
Any update on this? It's been nearly a year since v3.0.0 released.

**Comment by josecelano** [comment]
PR #2063 makes the requested feature one step closer: once it merges, strict parsing/validation and v3 schema support will be present. Production activation still awaits configuration schema v3.0.0 runtime wiring as part of configuration epic [#1978](https://github.com/torrust/torrust-tracker/issues/1978) / [#1980](https://github.com/torrust/torrust-tracker/issues/1980). The option will become usable once that epic is complete, and it will be released in the next major 4.0.0 release. It is not currently available in production. [PR #2063](https://github.com/torrust/torrust-tracker/pull/2063)
